### PR TITLE
feat: support LoRA LM Head

### PR DESCRIPTION
**Pull request by Dogacel**
Support low-rank approximation of LM Head.

Paper: **SlimSpec: Low-Rank Draft LM-Head for Accelerated Speculative Decoding**
https://arxiv.org/html/2605.10453v1

Add `lm_head_rank_divisor` to the draft model config. When set, the draft's output projection is stored as two matrices, `lm_head.down` [rank, hidden] and `lm_head.up` [vocab, rank] with rank = hidden // divisor, and the dense [vocab, hidden] product is never materialized.

image

Small training run on 20K data shows it has higher loss, but near frozen head acceptance.

### CODE REVIEW

File: torchspec/models/draft/base.py [hunk 1]
```diff
@@ -91,6 +92,43 @@ def prepare_decoder_attention_mask(
...
+
+    def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
+        return self.up(self.down(hidden_states))
+
+
+def build_lm_head(config, hidden_size: int, vocab_size: int, target_vocab_size: int) -> nn.Module:
+    """Dense head, or a ``LowRankHead`` when the config sets ``lm_head_rank_divisor``."""
+    divisor = getattr(config, "lm_head_rank_divisor", None)
+    if not divisor:
+        return nn.Linear(hidden_size, vocab_size, bias=False)
```
[the remark this answers is not in the document]
Comment (Dogacel) [reply torchspec/models/draft/base.py:123]: It's fine..

**Review by yubofredwang — APPROVED** [review]
